libraries/lib-xml/XMLTagHandler.cpp

```diff
@@ -170,39 +170,32 @@ bool XMLValueChecker::IsGoodInt64(const wxString & strInt)
 
 bool XMLTagHandler::ReadXMLTag(const char *tag, const char **attrs)
 {
-   wxArrayString tmp_attrs;
+   mCurrentTagAttributes.clear();
 
    while (*attrs) {
-      const char *s = *attrs++;
-      tmp_attrs.push_back(UTF8CTOWX(s));
-   }
+      const char* name = *attrs++;
+      const char* value = *attrs++;
 
-// JKC: Previously the next line was:
-// const char **out_attrs = NEW char (const char *)[tmp_attrs.size()+1];
-// however MSVC doesn't like the constness in this position, so this is now
-// added by a cast after creating the array of pointers-to-non-const chars.
-   auto out_attrs = std::make_unique<const wxChar *[]>(tmp_attrs.size() + 1);
-   for (size_t i=0; i<tmp_attrs.size(); i++) {
-      out_attrs[i] = tmp_attrs[i];
+      mCurrentTagAttributes.emplace_back(
+         std::string_view(name), XMLAttributeValueView(std::string_view(value)));
    }
-   out_attrs[tmp_attrs.size()] = 0;
 
-   bool result = HandleXMLTag(UTF8CTOWX(tag), out_attrs.get());
+   bool result = HandleXMLTag(tag, mCurrentTagAttributes);
```

Comment:
By eliminating UTF8CTOWX, are you requring that tag strings to be canonicalized into utf-8 already, or requiring them just to be ASCII strings?  They are all ASCII in practice (I just had to  review them all).  It is worth a comment.

You know I like to say things with type disctinctions where I can.  If we were using C++20, I would use std::u8string_view instead of std::string_view.  Not having C++20 and char8_t, what can we do?  Maybe at least make a different type alias to make the disctinction in function declarations, thought we won't have type checking by the compiler.